### DISCUSSION TITLE
How can I automatically run picom when I start i3?

### DISCUSSION THREAD

**Discussion by CoconutMacaroon** [i3/i3#4841] [Q&A] [answered]
By default, I have a lot of screen tearing when I use Firefox in i3. I downloaded picom and tweaked the settings a bit, and I've found that if I run it in a terminal in i3 (`pycom --config ~/.concif/picom/picom.conf`) the screen tearing goes away. I'd prefer to have it automatically run when I start i3. I'm using xorg, if it matters. How can I do that? I've tried adding `exec picom --config ~/.config/picom/picom.conf &` to the end of my `~/.config/i3/config` file, but that causes i3 to have a grey background at start and have a strange loading cursor, so I don't think that is correct.

So, what's the proper way to run that picom command when I start i3?

**Comment by stevelorenz** [comment, marked as answer]
Hi,
You can add `exec --no-startup-id picom --config ~/.config/picom/picom.conf -b` in `~/.config/i3/config`.
The `-b` option (or `--daemon`) will let picom run as a daemon process.

**Comment by RayZ0rr** [comment]
Or you can just use:
`exec --no-startup-id picom -b`
as picom automatically reads from picom.conf in the config folder.

**Comment by byteZorvin** [comment]
What happens if I dont add the -b option @stevelorenz

**Reply by Granth-Sudan** [reply]
Bro -b means baground

**Reply by Yuzuru-Yamazaki** [reply]
background

**Comment by YAUIO** [comment]
`exec_always --no-startup-id picom &`
and that's it

**Reply by fraubou1313** [reply]
Thanks, it works

**Reply by okraits** [reply]
You don't need the `&` at the end.

**Reply by YAUIO** [reply]
> You don't need the `&` at the end.

i mean yeah, but it won't ruin anything and i'm used to it when i launch something in the background
